### PR TITLE
[KIECLOUD-131] imageRegistry validation

### PR DESCRIPTION
**Pull request by tchughesiv**
Signed-off-by: tchughesiv <tchughesiv@gmail.com>

### CODE REVIEW

File: deploy/examples/image_registry.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: app.kiegroup.org/v1
+kind: KieApp
+metadata:
+  name: image-registry
+spec:
+  environment: rhpam-trial
+  imageRegistry:
+    insecure: true
+    registry: registry.example.com:5000
```
Comment (bmozaffa) [inline deploy/examples/image_registry.yaml]: See our validation [unit test,](https://github.com/kiegroup/kie-cloud-operator/blob/master/pkg/controller/kieapp/test/crd_validation_test.go#L26). I think we should extend it, or add another function for the examples folder.

Comment (tchughesiv) [reply deploy/examples/image_registry.yaml]: i already added it with a prior PR...
https://github.com/kiegroup/kie-cloud-operator/blob/master/pkg/controller/kieapp/test/crd_validation_test.go#L38

**Pull request merged**